### PR TITLE
test cleanup

### PR DESCRIPTION
**Pull request by theneva**
I cleaned up the tests according to @thomas11's review

### CODE REVIEW

**Comment by thomas11** [comment]
A little unfortunate that your commit is based on master, that conflates a bunch of changes with your actual test change. Any chance you could rebase on @julsemaan's PR branch?

File: provider/pkg/provider/util/testutil/test_ssh_server.go
Comment (theneva) [inline provider/pkg/provider/util/testutil/test_ssh_server.go:17]: I put this file in the testutil directory so it can be reused in the other test that depends on an SSH server.

I made that change here: https://github.com/theneva/pulumi-command/commit/bbae30e79444d8d28ad7a03c220c9cad9343d1da, and I'd love to include it as part of this PR. However, the test fails for me locally even on master, and I don't want to touch it (or block this PR on that unrelated change) since I can't verify the fix.

I'm happy to send that commit as a separate PR to pulumi/pulumi-command after this PR lands.

**Comment by theneva** [comment]
> A little unfortunate that your commit is based on master, that conflates a bunch of changes with your actual test change. Any chance you could rebase on @julsemaan's PR branch?

sorry @thomas11, I did change my mind and change it to be like you ask, but it looks like you're just too fast for me 😅 care to take another look now that the diff is smaller, please?

File: provider/tests/provider_test.go
Comment (thomas11) [inline provider/tests/provider_test.go]: Both of these tests update the same resource, right? Can they really run in any order then?

File: provider/pkg/provider/util/testutil/test_ssh_server.go
Comment (thomas11) [reply provider/pkg/provider/util/testutil/test_ssh_server.go:17]: That's a great move. I can help getting the other test to pass. It should pass on master in any case. What's the error you're getting?

File: provider/tests/provider_test.go
Comment (theneva) [reply provider/tests/provider_test.go]: Huh, I apparently only read half of your [comment](https://github.com/pulumi/pulumi-command/pull/451#discussion_r1636953415), and just made them run in parallel…

I didn't have any issues with the parallelism when running the tests locally, but I suppose you're right that they might get in each other's way.

Addressing your original comment on this: The create step does have an assertion that's relevant to the test, so I don't really view it as pure setup. Keeping it as part of the test – the way it is right now – makes the most sense to me.

Do you think this part looks ready to merge now?

File: provider/pkg/provider/util/testutil/test_ssh_server.go
Comment (theneva) [reply provider/pkg/provider/util/testutil/test_ssh_server.go:17]: Just to be clear: The new test I wrote passes on my machine just fine. I hope we can land this PR without figuring out why the `TestOptionalLogging` test fails for me!

Buuuut since you asked:

I get assertions failures that seem to indicate that the output from the SSH server just doesn't get registered by the context.

I made a quick recording, showing me running `go test ./...` from `./provider` on the latest `origin/master` (914454482a2634938009eb41915467f867a6d4d6):

https://github.com/julsemaan/pulumi-command/assets/1404650/e471639c-bc0f-4416-9412-7716e44e37ab

I'm running this Go (installed with homebrew):

```
$ go version
go version go1.22.3 darwin/arm64
```

… on this machine:

image

I also tried running `go install ./...` and `go mod tidy` just to be sure, but neither command did anything.

Again, I'm happy to help with figuring out why that test fails for me, but I'd really like to land this PR first 😅

File: provider/tests/provider_test.go
Comment (thomas11) [reply provider/tests/provider_test.go]: Yep, LGTM!

**Review by thomas11 — APPROVED** [review]

**Review by thomas11 — APPROVED** [review]

File: provider/pkg/provider/util/testutil/test_ssh_server.go
Comment (thomas11) [reply provider/pkg/provider/util/testutil/test_ssh_server.go:17]: I can reproduce this locally. But if you run the same `go test ./...` in `provider/tests`, it works. That's how the CI build passes.

Comment (thomas11) [reply provider/pkg/provider/util/testutil/test_ssh_server.go:17]: I can reproduce this and will look into it.

Comment (thomas11) [reply provider/pkg/provider/util/testutil/test_ssh_server.go:17]: The regression was introduced by `36798ab Upgrade to latest pulumi-go-provider v0.17 (#442)`.

The reason CI tests pass is that it runs `go test` in `provider/tests`, not `/provider` - needs to be fixed as well.

Comment (theneva) [reply provider/pkg/provider/util/testutil/test_ssh_server.go:17]: I see! That sounds like a fix for master. I'm happy to submit the change to use the new common test SSH server once it works on master

**Comment by theneva** [comment]
@julsemaan looks like we're good to go! If you agree, will you merge this PR? :smile:

File: provider/pkg/provider/util/testutil/test_ssh_server.go
Comment (thomas11) [reply provider/pkg/provider/util/testutil/test_ssh_server.go:17]: This is now fixed in master.

Comment (theneva) [reply provider/pkg/provider/util/testutil/test_ssh_server.go:17]: Awesome! By the way, are you able to merge this PR? If you can, I'm pretty sure Julien won't mind :smile:

**Comment by julsemaan** [comment]
> @julsemaan looks like we're good to go! If you agree, will you merge this PR? 😄

image

**Pull request merged**
